src/osd/PrimaryLogPG.cc

```diff
@@ -11535,13 +11560,22 @@ void PrimaryLogPG::prep_backfill_object_push(
 
   // We need to take the read_lock here in order to flush in-progress writes
   obc->ondisk_read_lock();
-  pgbackend->recover_object(
+  int r = pgbackend->recover_object(
     oid,
     v,
     ObjectContextRef(),
     obc,
     h);
   obc->ondisk_read_unlock();
+  if (r < 0) {
+    dout(0) << __func__ << " Error " << r << " on oid " << oid << dendl;
+    list<pg_shard_t> fl = { pg_whoami };
+    failed_push(fl, oid);
+    primary_error(oid, v);
+    backfills_in_flight.erase(oid);
+    missing_loc.add_missing(oid, v, eversion_t());
```

Comment:
this missing_loc.add_missing() update should just be part of primary_error() - the only caller that doesn't do it is prep_object_replica_pushes(), and that's a bug - it's not marked missing yet on the primary yet in that case, but should be